### PR TITLE
 include EOL distros in rosdep update

### PR DESCRIPTION
**Pull request by ipa-mdl**
Addresses https://github.com/ros-infrastructure/rosdep/pull/647
The flag will only be set for EOL distros on systems which support this flag.

### CODE REVIEW

**Comment by ipa-mdl** [comment]
Perhaps we want to inlcude an opt-out for users of EOL Docker images, which might not ship the new flag.

File: industrial_ci/src/tests/source_tests.sh [hunk 1]
```diff
@@ -79,8 +79,17 @@ rosdep --version
 if ! [ -d /etc/ros/rosdep/sources.list.d ]; then
     sudo rosdep init
 fi
-ret_rosdep=1
-rosdep update || while [ $ret_rosdep != 0 ]; do sleep 1; rosdep update && ret_rosdep=0 || echo "rosdep update failed"; done
+
+update_opts=()
+case "$ROS_DISTRO" in
+"hydro"|"jade")
+    if rosdep update --help | grep -q -- --include-eol-distros; then
```
Comment (130s) [inline industrial_ci/src/tests/source_tests.sh]: I found this `if` clause a thoughtful addition :).

In `else` situation for this `if`, I assume `rosdep` is old so all rosdep key still get fetched and no error occurs, thus we don't need `else` handling?

File: industrial_ci/src/tests/source_tests.sh [hunk 2]
```diff
@@ -79,8 +79,17 @@ rosdep --version
...
+      update_opts+=(--include-eol-distros)
+    fi
+    ;;
+esac
+
+ici_retry 2 rosdep update "${update_opts[@]}"
```
Comment (130s) [inline industrial_ci/src/tests/source_tests.sh]: Just curious, what happens without retrying here?

Comment (ipa-mdl) [reply industrial_ci/src/tests/source_tests.sh]: The retry was in there before..

Comment (ipa-mdl) [reply industrial_ci/src/tests/source_tests.sh]: There is a change though: now a failed update will stop the test and not just print a warning..

File: industrial_ci/src/util.sh [hunk 3]
```diff
@@ -172,12 +172,16 @@ function ici_enforce_deprecated {
...
 
   for ((i=1;i<=tries;i++)); do
     "$@" && return 0
+    ret=$?
     sleep 1;
   done
-  error "'$*' failed $tries times"
+
+  ici_color_output ${ANSI_RED} "'$*' failed $tries times"
+  return $ret
```
Comment (130s) [inline industrial_ci/src/util.sh]: https://github.com/ros-industrial/industrial_ci/pull/352/files#r251236099

> There is a change though: now a failed update will stop the test and not just print a warning..

I might be misunderstood, but doesn't this line and the subsequent code prevent the ICI from failing?

Comment (130s) [reply industrial_ci/src/util.sh]: @ipa-mdl This is me just trying to stay on the safest side. If my understanding above is wrong then we can go ahead (if so please merge) as CI is passing.

Comment (ipa-mdl) [reply industrial_ci/src/util.sh]: This line returns a non-zero code, which is treated as error in the calling code.

**Pull request merged**